File: tests/hashing-match.js
var BruteForcer = require('../BruteForcer');

var bf = new BruteForcer();

var d1 = +new Date;

hasher = str => {
    arr = str.split('');
    out = 1;
    arr.forEach(c => {
        out += out * c.charCodeAt(0)
    });

    return out;
};

if (bf.match(14251497120, hasher)) {
    var diff = +new Date - d1;
    console.log('Successfully matched \'hello\' with hashing function in', bf.roundcount, 'rounds (' + diff + 'ms)');
}

File: tests/hashing-match.py
import sys, os, time
sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
from BruteForcer import BruteForcer

bf = BruteForcer()

d1 = int(round(time.time() * 1000))

def hasher(str):
    arr = list(str)
    out = 1
    for i in xrange(0, len(arr)):
        out += out * ord(arr[i])

    return out

if (bf.match(14251497120, hasher)):
    diff = str(int(round(time.time() * 1000)) - d1)
    print "Successfully matched 'hello' with hashing function in " + str(bf.roundcount) + " rounds (" + diff + "ms)"